code/modules/antagonists/traitor/objectives/final_objective/battlecruiser.dm

```diff
@@ -0,0 +1,54 @@
+/datum/traitor_objective/final/battlecruiser
+	name = "Reveal Station Coordinates to nearby Syndicate Battlecruiser"
+	description = "Use a special upload card on a communications console to send the coordinates \
+	of the station to a nearby Battlecruiser. You may want to make your syndicate status known to \
+	the battlecruiser crew when they arrive, their goal will be to destroy the station."
+
+	///checker on whether we have sent the card yet.
+	var/sent_accesscard = FALSE
+
+/datum/traitor_objective/final/battlecruiser/generate_ui_buttons(mob/user)
+	var/list/buttons = list()
+	if(!sent_accesscard)
+		buttons += add_ui_button("", "Pressing this will materialize an upload card, which you can use on a communication console to contact the fleet.", "phone", "card")
+	return buttons
+
+/datum/traitor_objective/final/battlecruiser/ui_perform_action(mob/living/user, action)
+	. = ..()
+	switch(action)
+		if("card")
+			if(sent_accesscard)
+				return
+			sent_accesscard = TRUE
+			podspawn(list(
+				"target" = get_turf(user),
+				"style" = STYLE_SYNDICATE,
+				"spawn" = /obj/item/card/emag/battlecruiser,
+			))
+
+/proc/summon_battlecruiser()
+	var/list/candidates = poll_ghost_candidates("Do you wish to be considered for battlecruiser crew?", ROLE_TRAITOR)
```

Comment:
Make this its own preference, this is exactly the type of thing I split some prefs off to avoid

Reply:
@tralezab